pandas/core/frame.py

```diff
@@ -3821,13 +3821,16 @@ def _apply_standard(self, func, axis, ignore_failures=False, reduce=True):
         # e.g. if we want to apply to a SparseFrame, then can't directly reduce
         if reduce:
 
-            try:
-
-                # the is the fast-path
-                values = self.values
-                dummy = Series(NA, index=self._get_axis(axis),
-                               dtype=values.dtype)
+            # the is the fast-path
+            values = self.values
+            # Create a dummy Series from an empty array
+            # Unlike filling with NA, this works for any dtype
+            index = self._get_axis(axis)
+            empty_arr = np.empty(len(index), dtype=values.dtype)
+            dummy = Series(empty_arr, index=self._get_axis(axis),
+                           dtype=values.dtype)
```

Comment:
The problem is the previous code couldn't create an empty series of ints. I guess I could make this into a `Series.empty_like` class method and add tests for that, then replace this block with a single call to that method.